src/components/Profile/components/Identification/index.js

```diff
@@ -427,6 +429,12 @@ const Identification = ({ profile, myProf, network, createSnackbar }) => {
     setHasNickname(FirstName !== profile.NickName && profile.NickName !== '');
   }
 
+  function createMaidenName(profile) {
+    let Name = String(profile.fullName);
+    let LastName = Name.split(' ')[2];
+    setHasMaidenName(LastName !== profile.MaidenName && profile.MaidenName !== '');
```

Comment:
I know you were modeling this off the `createNickname` function, but this can be done a lot more simply. For one thing, `profile` already has `FirstName` and `LastName` properties (which is what we use to create `fullName`).

Also, `optional chaining` is useful here too.

```suggestion
    setHasMaidenName(profile?.MaidenName && profile?.LastName !== profile.MaidenName);
```

You could also apply this same refactoring `createNickname`.